### ISSUE TITLE
libspatialite filename wrong on macOS

### ISSUE THREAD

**Issue by mattcen** [sqlitebrowser/sqlitebrowser#2807]
Details for the issue
--------------------

#### What did you do?

```sh
brew install db-browser-for-sqlite libspatialite
```

Then, in DB Browser, try to Load Extension, and browse to `/usr/local/Cellar/libspatialite/5.0.1_1/lib` and try to load `libspatialite.dylib` or `mod_spatialite.dylib` (I forgot which one I needed so tried both).

Then I tried to load a database.

#### What did you expect to see?

The extension loading successfully and allowing geospatial queries

#### What did you see instead?

> Error loading extension: dlopen(/usr/local/Cellar/libspatialite/5.0.1_1/lib/mod_spatialite.7.dylib.dylib, 10): image not found

Note that it says `.dylib.dylib`; I think it's appending an additional `.dylib` to the end on top of the one that is already there.

If I create an empty file called `mod_spatialite` and try to load *that* as an extension, I instead get:

> Error loading extension: dlopen(/usr/local/Cellar/libspatialite/5.0.1_1/lib/mod_spatialite.dylib, 10): no suitable image found.  Did find:
> 	/usr/local/Cellar/libspatialite/5.0.1_1/lib/mod_spatialite.dylib: code signature in (/usr/local/Cellar/libspatialite/5.0.1_1/lib/mod_spatialite.dylib) not valid for use in process using Library Validation: mapped file has no cdhash, complete

Which now seems to be loading the correct file, but gives an error. I'm *pretty* sure this workaround worked for me in the past when I was using libspatialite 5.0.1 as opposed to 5.0.1_1, so maybe there's an incompatibility issue at play too.

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [x] macOS: (11.5.2)

#### What is your DB4S version?

- [x] 3.12.2

#### Did you also

- [x] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds

Yes, I *did* try this, and it actually seems to work, but here's the thing: I've had this issue before, and it seemed to get better when I tried an *earlier* nightly version (a version *before* 3.12.2), so I don't know whether the this issue occasionally regresses or what.

- [x] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by michalisa** [comment]
I am also encountering the same issue on this specific version, while it was working on previous versions. Running macOS as well. I do not think I can identify a workaround just yet.

**Comment by davenquinn** [comment]
I just tried to work around this by editing the path directly in the plist file in `Library/Preferences`. Unfortunately removing the `.dylib` extension did not help the application actually resolve the extension file. This is in version 3.12.2; extension loading was working properly in earlier versions.

**Comment by davenquinn** [comment]
Just trying some new workarounds for this, I had the bright idea to actually soft-link and/or copy the file `mod_spatialite.7.dylib` to `mod_spatialite.7.dylib.dylib` to see if the app successfully picks that up.

Instead I got the error message "'/usr/local/Cellar/libspatialite/5.0.1_2/lib/mod_spatialite.7.dylib.dylib' not valid for use in process: mapped file has no cdhash, co" which seems to be truncated.

Thought it would be useful to record that here.

**Comment by justinclift** [comment]
Sounds potentially useful, but yeah... I have no idea what what means either. :man_shrugging:

**Comment by chrisjlocke** [comment]
"...has no cdhash, completely unsigned? Code has to be at least ad-hoc signed."
/blinks blankly
Means nothing to me, but thought I'd fill in the blanks.

**Comment by justinclift** [comment]
> Code has to be at least ad-hoc signed.

Oh.  That bit sounds like it wants the library signed with a Apple Developer's certificate.  Probably in the same way - or similar to - how we need to sign our macOS releases.

**Comment by prashis** [comment]
I finally got it working for 3.12.99 version. I'd to do some workaround tho
- copied the mod_spatialite.dylib file w/o extension
```bash
brew install libspatialite && cp /usr/local/Cellar/libspatialite/5.0.1_2/lib/mod_spatialite.dylib /usr/local/Cellar/libspatialite/5.0.1_2/lib/mod_spatialite
```
- selected the `mod_spatialite` in extensions tab & removed the copy file `rm mod_spatialite`
- finally loaded the extension executing following SQL
```sql
SELECT load_extension('mod_spatialite');
```

**Comment by davenquinn** [comment]
Yes, I just tested this with the latest nightly (SQLCipher version, the other one doesn't seem to open the application) and the extension (which I load on startup from preferences) is loaded on app startup. Thanks for the fix!

**Comment by davenquinn** [comment]
OK, this bug appears to be back in the latest nightly? Just got this error message again upon update. I tested both the SQLCipher and vanilla versions; both appear to have the issue.

![image](https://user-images.githubusercontent.com/1537910/205467873-21846771-c887-493d-aea0-3617804ec313.png)

**Comment by davenquinn** [comment]
One significant change to my system is that I have upgraded recently to Ventura (this, incidentally, required updating the referenced Spatialite/proj libraries, but I have verified that the `spatialite` CLI tool works). Ventura changes explain the new `/System/Volumes/Preboot/Cryptexes/OS/...` prefix, although I don't know how exactly, if at all, that is involved in this problem...

**Comment by glelk** [comment]
This temporary solution worked for me: ad-hoc signing **both** the app and the dylib:

```
codesign --force --deep -s - /Applications/DB\ Browser\ for\ SQLite.app
codesign --force -s - extension_name.dylib
```

**Comment by justinclift** [comment]
@glelk Did you need to re-sign the .app, as well as the extension .dylib?

That seems a bit weird, as I'm still getting several emails each day from Apple's notarisation service about "Your signing was successful" (or wording along those lines).

That's coming from our overnight nightly build process, so I'm thinking the signing is still working. (?)
